v3.5.0.9 Beta

Die4Ever opened the pull request:
# REMINDER:

If you want to start with Speed Enhancement augmentation (improves running speed and jumping abilities) like previous versions, then select the loadout called "Speed Enhancement". Otherwise you will start with a new augmentation called "Running Enhancement" which only improves your running speed, with a chance to find "Speed Enhancement" in an aug can.

# Changes Since v3.5.0.8 Beta:

- Don't randomize name of M02 downstairs terrorist with memes disabled
- Better singular goal descriptions
- Make the Revision Graveyard crypt accessible again
- Graveyard datacube for the EM Emitter keypad can no longer be located behind the bookcase, in Dowd's safe, or (in Revision) the 
hidden underground area
- Vandenberg Computer electrical panels bingo goal is now "There's very little risk" and does not scale.  This is so it fits better on the bingo viewer and it's funnier.
- Improved Bingo Viewer display of goals with numbers
- Paris Club EmergencyExit bingo trigger
- Kill Tiffany Savage requires killing her yourself (and renamed to "Give it your best shot")
- Various other minor bingo goal name changes to make them fit on the bingo viewer better
- Added bingo goals for losing arms and legs, and losing limbs in general
- Metric Units will use km/h instead of m/s for "long distance" measurements (aka movement speed)
- smarter checks for banning MedicalBot_ClassDead/RepairBot_ClassDead vs UtilityBot_ClassDead
- prevent Tim's Closet Key from being inside the closet

# Previously: https://github.com/Die4Ever/deus-ex-randomizer/pull/1223#issuecomment-2902814398

Die4Ever commented on src/DXRModules/DeusEx/Classes/DXREvents.uc:
```diff
@@ -2078,6 +2082,22 @@ simulated function _CreateBingoBoard(PlayerDataItem data, int starting_map, int
 
     if(dxr.flags.settings.merchants < 20) data.BanGoal("DXRNPCs1_PlayerDead", 1);
 
+    if (dxr.flags.settings.medbots < 15 ||
```
-1 is for vanilla, which probably should not ban these goals

theastropath replied:
Ah yeah, true (or at least, it should be appropriately considered to have "enough bots")

Die4Ever replied:
```
simulated function _CreateBingoBoard(PlayerDataItem data, int starting_map, int bingo_duration, optional bool bTest)
{
    local int starting_mission, end_mission, real_duration;
    local int medbots, repairbots;

    starting_mission = class'DXRStartMap'.static.GetStartMapMission(starting_map);
    end_mission = class'DXRStartMap'.static.GetEndMission(starting_map, bingo_duration);
    real_duration = class'DXRStartMap'.static.SquishMission(end_mission) - class'DXRStartMap'.static.SquishMission(starting_mission) + 1;

    medbots = dxr.flags.settings.medbots;
    if(medbots <= -1) medbots = 30;
    medbots *= real_duration;

    repairbots = dxr.flags.settings.repairbots;
    if(repairbots <= -1) repairbots = 30;
    repairbots *= real_duration;

    // https://github.com/Die4Ever/deus-ex-randomizer/issues/1095
    if(medbots < 20 || repairbots < 20) {
        data.BanGoal("MedicalBot_ClassDead", 1);
        data.BanGoal("RepairBot_ClassDead", 1);
    } else {
        data.BanGoal("UtilityBot_ClassDead", 1);
    }

    if(dxr.flags.settings.merchants < 20) data.BanGoal("DXRNPCs1_PlayerDead", 1);

    if (medbots < 20 ||
#ifndef hx
        dxr.flags.settings.CombatDifficulty > 6 ||
#endif
        dxr.flags.settings.medkits < 50 ||
        dxr.flags.settings.health < 75) {
        data.BanGoal("JustAFleshWound", 1);
        data.BanGoal("LostLimbs", 1);
    }

```

Die4Ever commented:
# REMINDER:

If you want to start with Speed Enhancement augmentation (improves running speed and jumping abilities) like previous versions, then select the loadout called "Speed Enhancement". Otherwise you will start with a new augmentation called "Running Enhancement" which only improves your running speed, with a chance to find "Speed Enhancement" in an aug can.

# OVERALL CHANGES SINCE v3.4.1

# Goals Rando:

- Options for always-on Goal Location Hints (like Speedrun Training Mode) or Goal Location Spoilers
- In-world markers for non-spoiler "Goal Locations" button, similar to "Show Spoilers"
- Fixed player rotation for random start locations not being applied
- M02 NYC
  - M02 Battery Park start locations rework
  - M02 Battery Park Ambrosia gives more skill points to make it more worth searching for
  - M02 warehouse goals rando, more locations and more mutual exclusions
- M03 NYC
  - M03 Battery Park removed vent start
- M05 Jail
  - Moved M05 UNATCO Bathroom goal location to face a urinal
- M06 Hong Kong
  - Opening DTS sword case instantly plays infolink and gives skill points, so it's more obvious you don't need to wait
  - New Gordon Quick locations
    - The DTS and Gordon won't be on the same map.
    - If one is in market, the other is in Tonnochi. (However, canals+Tonnochi is also possible, or Tonnochi+Lucky Money)
  - Versalife radiation controls computer can no longer spawn in the overlook office
- M08 Return to NYC
  - Slightly moved Free Clinic goal location for better visibility
  - The flaming barrel near Dowd will no longer be moved, it's good for visibility
- M09 Ship Below Decks
  - Don't allow too many goals in engine room or pumps room
  - Require at least 1 goal near helipad
  - Multiple smaller BlockMonsters for Superfreighter weld points
  - Ship weld point moved Bilge Pumps Balcony so you can't destroy 2 weld points with 1 explosive
  - Ship weld points now slightly brighter
- Vandenberg
  - Vandenberg Cmd keypads change color when activated, so you don't accidentally double press them
  - Vending machine for unlucky Vandenberg Command goals rando
  - Added two new locations for Walton Simons in the Vandenberg Sub Base, to make meeting him in Area 51 more common
  - Silo: no more than 1 goal in each sector
    - courtyard inside the fence
    - Launch Command room
    - Silo tube
    - around the launchpad outside of the fence
- Area 51
  - Added new Sniper Tower starting location to Area 51, fixed double chance of vanilla start location
  - 15_AREA51_PAGE more goals rando spots
    - look inside the rooms with the switches for closing the UCs
    - Aquinas Substation Computer cannot be in the top floor UC control room

# Augmentations:

- Run Silent is now always 0% noise, the strength adjusts your speed
  - Level 1 is a default of 50% movement speed (crouch is 30%)
  - Level 4 is a default of 100% movement speed (speed is unaffected when turning on and off)
  - Can be boosted to level 5 for a default of 120% speed compared to regular running
  - I think this is much more intuitive than the vanilla Run Silent
- New leg augs:
  - Running Enhancement
  - Jump Enhancement, is automatic
  - (the old leg augs still exist separately: Speed Enhancement, Run Silent, and Ninja Aug)
- Leg aug strengths are randomized in the new "Strong Augs" mode
- New eye augs (only cost a small amount of energy):
  - Short-Range Vision Enhancement
  - InfraVision
  - Motion Sensor
- EMP Shield now protects from Shocked damage, and has better auto behavior, and max level is always 100% resistance
- Fixed Infravision highlighting movers and clouds
- Motion Sensor aug highlight now scales with velocity
- Tweaks to vision augs and fixed stacking
- Icons for new vision augs
- Aug description for Spy Drone actually provides useful info
- Improved names shown with Targeting aug. Add default familiar and unfamiliar names for Merchant and LeMerchant

# Loadouts:

- Most loadouts now start with Running Enhancement instead of Speed Enhancement
  - Speed Enhancement, Jump Enhancement, and Run Silent can also be found in aug cans
  - **Added new Speed Enhancement loadout**, which is the same as the old "All Items Allowed" loadout **(recommended for speedrunners)**
- The Three Leg Augs loadout
  - Speed Enhancement is banned
  - Running Enhancement, Jump Enhancement, and Run Silent are allowed
  - You start with one of the 3 leg augs at random (except the old Speed Enhancement)
  - Enables aug slot rando by default
- My Vision Is Augmented loadout
  - Allows Vision Enhancement, Short-Range Vision Enhancement, InfraVision, and Motion Sensor
  - You start with one of the 3 eye augs at random (except the old Vision Enhancement)
  - Enables aug slot rando by default
- SWTP now starts with InfraVision augmentation (temporary change?), and allows Jump Enhancement augmentation to be found
- Items spawned for loadouts appear in small crates. Extra items spawned in Warehouse now spawn in crates instead. Mission 3 entrance rando rebreathers spawn in crates. This should improve visibility.
  - Reduced loadouts item spawn rates due to crates being more visible
- Starting augs given again on New Game+
- Don't take away loadout starting item or rubber baton on NG+
- Ban weapon mods in Grenades Only loadout (they're useless)

# New Game Modes:

- New Speedrun Training game mode
  - Teaches you the possible goal locations without spoiling them.
- Strong Augs game mode - makes augs stronger

# Major Changes:

- WaltonWare speedrun splits overlay
  - In the Advanced settings or In-Game Flags set Splits Overlay to Show
- Now tracking the time of all previous loops in NG+ (in total, not separately)
- Option to use metric units!
- In-flight projectiles get deleted pretravel, to help prevent softlocking with inevitable damage traveling back to a map
- Area 51 Entrance fixed keys/datacube rules, and added another placeholder item spot
- New placeholder item locations in Area 51 Page, new datacube rules for 15_Datacube11, which has the code to the Aquinas Substation and Router Control Room.
- Area 51 Final important datacubes can no longer spawn in the aug upgrade container
- Allowed keybinding to multiple actions and ask before overwriting hotkey
- Enemies with gas grenades or pepper spray now always have visors 
  - Otherwise visor chance has been reduced from 33.3% to 30% to offset that
- Gas grenade clouds now last 22 to 27 seconds, instead of 30 to 40
- Enemy sidearms (pistols and crossbows), and crossbows have 50% chance of non-tranq darts
- Buffed plain darts from 15 damage to 17
- Minimum damage of 2 for tranq darts and other projectiles
- Slightly more pistol ammo
- Added electrical panel near electricity box in Catacombs Tunnels when repair bots are randomized
- Improved shuffle algorithms
  - (savegame compatibility issue if you're in the middle of M06, M08, or M10, safest time to update is when Jock is ready to take you to the next area)
- WaltonWare and NG+ crates
- New option to either pick the last used JC skin or randomly select one every time you load the new game menu (or continue to default JC every time).
- In-Game help buttons for game modes, loadouts, difficulties, autosave settings, and more!
- Fixed weapon MaxRange with range mods not carrying over across levels
- Dockyard rooftop vent always breakable with min damage of 1
- Unhackable keypads becoming hackable now have generally lower strength
- Falling platforms in M03 Airfield Helibase are only disabled in modes that need backtracking (bingo modes and entrance rando)
- DuClare basement key will only be on the second floor
- Compass direction gets mirrored when appropriate so verbal directions match the on-screen compass
- The Merchant can properly transfer items from packages that aren't the DeusEx package (which broke Rubber Baton sales in non-vanilla)
- Show next loop starting map when bingo_win is greater than zero in the Bingo Info window
- Allied Pawns in Standing or Dancing state will now move out of the way if you bump into them while holding a weapon
- Pawns who fail to pathfind for Patrolling state will now switch to Wandering state instead of Standing
- M09 starts gives ship maps
- Give Vandenberg Control Room key when startng in Vandenberg Computer/Control Room and later
- Installer now creates shortcuts on desktop and in start menu (can be disabled in Advanced)
- Added delete autosaves button (won't detect old autosaves from before this version, delete those manually)

# Minor Changes:

<details>
<summary><h2>Click to expand minor changes</h2></summary>

- Fixed crash when skipping ending too quickly
- More locations for M03 Airfield Dormitory Key (the whole area past the east gate, except for the security tower and the dormitory itself)
- Slightly toned down CombatDifficultyMultEnviro() (plasma damage against the player won't scale as much with difficulty)
- Fixed the softlock fix for conversation before M04 Hotel Ambush so that it actually plays the correctly gendered conversation
- Bob actually comments on your health on the elevator from Area 51 sector 2 to 3, along with a Mastodon toot
- Spawning new enemies checks for free space, to prevent blocking hallways
- Conversation with Paul at the end of M04 can now be initiated by Frob (as long as you've entered the flagtrigger near him)
- Added a 600 credit price limit to Le Merchant's item selection, so he won't have camo anymore
- Fixed M04 floating cloned enemies outside after the raid, should also affect M06 and M08 raids
- First person conversations (barks) no longer block WaltonWare wins, and don't count as InGame anymore
- Fixed M02 hotel hostage situation AI
- Spider Bot doors in HK Level 2 labs can't be closed with the keypad again
- OceanLab UC Spiderbots run towards the laser when triggered instead of attacking the player directly
- Silo ambushers will now run down the tunnel and back instead of directly attacking the player
- Added "destroy utility bots" bingo goal when medbot/repairbot chances are low
- If The Merchant chances are low, ban the bingo goal for killing him
- Selling Zyme to Renault for 65 credits now also counts towards bingo
- Don't randomize throwing knives in Anna's Mannequin in Revision
- Make sure FlyGenerator's get moved with their parent decoration
- Prevent duplicate Jojos
- Highlighting a modded weapon now shows that it is modified
- Revision player move speed now changed by AugNinja and AugOnlySpeed
- New icon for Ninja Aug
- Increase size of triggers for rescuing catacombs hostages (so that they can't dodge them)
- Computer login screens no longer clear the username if login fails
- Add Random button under portrait on New Game screen to select a random JC
- Fixed InHand item in Idle2 state in AnyEntry
- Empty InformationDevices are not shuffled (helps with Revision, where there are often decorative datacubes or newspapers in inaccessible areas)
- proper rounding of range values for weapons with nerfed ranges
- Extra placeholder locations for Hong Kong Helibase
- Show default skill costs
- Fixed some lamps starting half-on without autolamps
- Make throwing knives way cooler
- Thick spicy farts
- Crowd Control Nudge works more like the Superfreighter shaking
- Fixed M03 Airfield Helibase enemies calling the elevator
- Items no longer get added to your belt automatically if you already have another item of the same class on your belt
- Upgrading level 3 augs while synth heart is enabled
- HK Helibase now allows the Flight Deck Key and Master Locker Key to get swapped to the rooftop
- HK Level 2 Labs lockdown door only closes after destroying the UC, instead of when uploading the virus, to ensure there is always an exit available.
- Fixed Scott's name
- Fixed some door fragment types for normally unbreakable doors
- Fixed some double spaces
- Minimized leftover laser sight dots at level transitions when autolaser is enabled
- Stopped cloning mechanics in M12 comms building (Mike and Larry), which fixes in-fighting at Vandenberg Command
- Removed collision for OrdersTrigger at cathedral that disables the military bot patrols
- Cutscene dialog only randomizes within the set of conversations matching the selected gender (This prevents duplicate lines from coming across from the opposite gender conversations, or JC speaking with the wrong voice)
- Simplified menu options for doors randomization
- Added support for timed races (Silo's Secret Slide, Chateau key hunt, etc) and Mastodon toots for them
- Area 51 UC Door Keypads open the door instead of just unlocking them
- Fixed "In-Game Settings" menu setting combat difficulty
- Fixed hole in wall outside UNATCO HQ M03
- Removed StopWhenEncroach doors from UNATCO HQ, so they're less janky
- Randomized names no longer in Zero Rando even with memes enabled (so people don't get confused)
- Computer Hack button is no longer clickable when you have no energy (and the balance change is enabled)
- "Other" enemy faction can no longer spawn grays, meaning there won't be friendly grays anymore
- Added some additional context to conversation options with Harley in M03 Battery Park to explain why you are unable to do anything but say not interested to him.
- Added option to disable death hints
- Black vans now block AI vision so you can hide behind them
- Loadout Crates get randomized rotation
- Moved LDDP Russ slightly to the side at the Lucky Money front door, male vs female JC races are now fair!
- Fixed MJ12 in Maggie's living room (up to 500% enemies)
- Removed large unbreakable crates from M05 UNATCO HQ, because they're annoying
- Crowd Control Pianos drop with a randomized yaw, so they aren't all facing the same direction
- Added spawn limits to Spam Datacubes, Marbles, and Pianos for Crowd Control
- Trigger All Alarms Crowd Control effect no longer works on disabled or confused alarm things, or friendly cameras
- DXVK v2.6.1
- Improved The Floor Is Lava Crowd Control effect
- Float Away Crowd Control effect no longer gets applied to hidden decorations (which stops the M02 Warehouse generator explosive crates from floating away)
- Switch lasers in the tunnel towards Silo computer room to Beam Triggers that set off the alarm unit next to them (instead of being Laser Triggers that act as their own alarm)
- Hacked computer account switching window appears even on the special options screen.  Also make sure the known accounts window disappears in Revision when a computer is hacked.
- Fixed losing autosave when losing flags
- Corpse Explosion CC effect won't blow up invincible carcasses
- Removed the random Thug on M05 Liberty Island (So Crowd Control can't swap the player with him)
- CarWrecked bBlockSight
- Fixed AugNinja for VMD
- All scriptedpawns in cutscenes are vincible except for the speaking ones, and more cutscene nonsense
- Randomized location of fight during intro
- Numpad number keys can select the cameras on a security computer. The row of numbers over the letters already does this.
- Disable autorun on pretravel
- Door frobbability and highlightability copied to matching doors (fixes Versalife labs elevator doors)
- Prevent UNATCO corpse from reanimating when reanimation is enabled outside of Halloween Mode
- Graveyard zombies dig out of the ground and generate dirt fragments
- Extra Mastodon messaging for Horde Mode
- Horde Mode now places spawned items into crates
- Templar Computer in Horde Mode gets shuffled every wave and gives 500 skill points when it's used each wave.  Ban weapon mods in Freeman Mode and SWTP Pure loadouts.
- Actually play the prod sound when Horde Mode shocks you
- Fixed appearance of rando-specific enemies in VMD
- Update damage multiplier logic for RevRandoPlayer to match vanilla
- Epilepsy Safe lighting can be toggled on and off instantly
- Horde Mode reports remaining enemies less frequently, but reports immediately if the number of remaining enemies changes.
- The M02 Free Clinic cabinet key can't be used to open drawers in M05
- Added PlaceholderItem to the dead end tunnel at the end of catacombs
- Don't allow "KnowYourEnemy" goal on 41+ starts. Allow it on a 36 start
- Removed "throw melee" option
- Most Kill/Knock Out bingo goals require that the player's the one who does it
- Fixed kill/knock out Anna goals (and maybe others)
- You can now stand on WHChairDining
- Fixed scroll position for In-Game Flags menu on init
- Fixed shotguns firing rate descriptions
- Looting ammo from a weapon from a carcass now properly sets the owner. (May fix some strange bugs with inventory)
- Lucky Money start plays infolink to avoid confusion (Good. Now take the sword to Max Chen at the Lucky Money Club.)
- Reduced hack strength on M06 helibase electric panel
- Fashion now respects the players skin tone. Fem JC can also no longer go topless while wearing a trench coat.
- Unaugmented Prison Pocket
- Paul's apartment door in M04 can be manually closed again (It works this way in M02)
- Always RemoveStopWhenEncroach() movers because it's bad
- M15 Bunker PlaceholderEnemy on top of vent to make that route slightly harder and more in-line with the blast door route
- Update rotations on all randomized start locations so they're facing an identifiable direction (instead of a wall or something)
- Show fire rate of melee weapons on Inventory screen.
- Replaced Vandenberg comsat random start with tunnels (basically the same thing)
- M06: storage UC_Chamber_Door not breakable (breaking this can cause a lot of confusion, and you need to use the computer anyways to get the UC schematics in order to progress)
- Prevent M05 Karkians from being cloned, so they don't clone outside the enclosure
- Zero Rando fixed CombatDifficulty being slightly off when below Realistic
- Clothes Looting setting persists through NG+
- Floor Is Lava: fix ladder check for Revision
- Create new Training Merchant for use in... Training.
- WaltonWare starts later than Vandenberg Command now properly destroy the entry jock
- WaltonWare start later than Vandenberg Command, Comms door cannot be closed
- Hide bingo board during intro so you can't cheat races that are using in-game time
- Remove different color for bingo goals without a mission mask
- Improved Bingo Viewer display of goals with numbers
- Various other minor bingo goal name changes to make them fit on the bingo viewer better
- Added bingo goals for losing arms and legs, and losing limbs in general
- Shoot Satellite Dishes at Sub Base bingo goal now has better bingo trigger collision
- Added Paris Club EmergencyExit bingo trigger
- Kill Tiffany Savage requires killing her yourself (and renamed to "Give it your best shot")
- Rando camera view distance reduced to match beep sound distance
- Slightly quieter sawed off fire sound and plasma impact sound
- Dress and SuitSkirt models share leg textures for fashion purposes
- In-Game Flags menu disables buttons for goals rando and starting locations options for certain maps
- Paul doesn't leave the hotel after the M04 raid until you leave.  Make the flag setting when leaving the hotel (during or after the raid) a bit more robust, hopefully.
- Track body part losses for vanilla and Revision
- Merchant conversation flags are now unique per map and bindname
- Race timers for training missions
- Many important NPCs will now return to their original locations after fleeing
- Better singular goal descriptions
- Made the Revision Graveyard crypt accessible again
- Graveyard datacube for the EM Emitter keypad can no longer be located behind the bookcase, in Dowd's safe, or (in Revision) the 
hidden underground area
- Horde mode:
    - Horde Mode no longer bans skills outright, uses aug slot rando, and doesn't reroll skill costs
    - Allow Ironman (All Saves Disabled) save mode to be shown in the list if it is the current default
    - Remove forced Speed Enhancement from Horde Mode, now it depends on your loadout where Speed Enhancement is still selectable
    - Added 20mm, Sabot, and WP to item options.

![image](https://github.com/user-attachments/assets/59d303ab-f61c-402b-8e0f-abf78b2a0848)
</details>
